src/simsopt/field/coil.py

```diff
@@ -185,3 +186,108 @@ def coils_via_symmetries(curves, currents, nfp, stellsym):
+    """
+    Export a list of Curve objects together with currents in MAKEGRID format, so they can 
+    be used by MAKEGRID and FOCUS. The format is introduced at
+    https://princetonuniversity.github.io/STELLOPT/MAKEGRID
+
+    Args:
+        filename: Name of the file to write.
+        curves: A python list of Curve objects.
+        currents: Coil current of each curve.
+        groups: Coil current group. Coils in the same group will be assembled together. Defaults to None.
+        nfp: The number of field periodicity. Defaults to 1.
+        stellsym: Whether or not following stellarator symmetry. Defaults to False.
+    """
+
+    assert len(curves) == len(currents)
+    assert isinstance(curves, CurveXYZFourier)
+    coils = coils_via_symmetries(curves, currents, nfp, stellsym)
+    ncoils = len(coils)
+    if groups is None:
+        groups = np.arange(ncoils) + 1
+    else:
+        assert len(groups) == ncoils
+        # should be careful. SIMSOPT flips the current, but actually should change coil order
+    with open(filename, "w") as wfile:
+        wfile.write("periods {:3d} \n".format(1))  # force to write nfp=1 for now
+        wfile.write("begin filament \n")
+        wfile.write("mirror NIL \n")
+        for icoil in range(ncoils):
+            x = coils[icoil].curve.gamma()[:, 0]
+            y = coils[icoil].curve.gamma()[:, 1]
+            z = coils[icoil].curve.gamma()[:, 2]
+            for iseg in range(len(x)):  # the last point match the first one;
+                wfile.write(
+                    "{:23.15E} {:23.15E} {:23.15E} {:23.15E}\n".format(
+                        x[iseg], y[iseg], z[iseg], coils[icoil].current.get_value()
+                    )
+                )
+            wfile.write(
+                "{:23.15E} {:23.15E} {:23.15E} {:23.15E} {:} {:10} \n".format(
+                    x[0], y[0], z[0], 0.0, groups[icoil], coils[icoil].curve.name
+                )
+            )
+        wfile.write("end \n")
+    return
+
+
+def coils_to_focus(filename, curves, currents, nfp=1, stellsym=False, Ifree=False, Lfree=False):
+    """
+    Export a list of Curve objects together with currents in FOCUS format, so they can 
+    be used by FOCUS. The format is introduced at
+    https://princetonuniversity.github.io/FOCUS/rdcoils.pdf
+
+    Args:
+        filename: Name of the file to write.
+        curves: A python list of Curve objects.
+        currents: Coil current of each curve.
+        nfp: The number of field periodicity. Defaults to 1.      
+        stellsym: Whether or not following stellarator symmetry. Defaults to False.
+        Ifree: Flag specifying whether the coil current is free. Defaults to False.
+        Lfree: Flag specifying whether the coil geometry is free. Defaults to False.
+    """
+    from simsopt.geo import CurveLength
+
+    assert len(curves) == len(currents)
+    assert isinstance(curves, CurveXYZFourier)
```

Comment:
Think you should be asserting individually for each element in `curves` here, since `curves` is a list according to the docstring.

Reply:
You are right. The assertation has been moved inside the `for` loop.